src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_NUMBER = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void growIfArrayFull() {
+        if (size >= array.length) {
+            int newCapacity = (int) (array.length * GROWTH_NUMBER);
+            if (newCapacity <= size) {
+                newCapacity = size + 1;
+            }
+            array = Arrays.copyOf(array, newCapacity);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Such index can not be add" + index);
+        }
+        growIfArrayFull();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Such index don not exist");
```

Comment:
Add index with whitespace as you should have done in method add:
`throw new ArrayListIndexOutOfBoundsException("Such index dont exist: " + index);`